### PR TITLE
Feature/tao 7961/implement test state api

### PR DESCRIPTION
**Pull request by ihar-aleksiaichuk**

### CODE REVIEW

File: controller/Results.php [hunk 1]
```diff
@@ -550,4 +553,63 @@ public function export()
 
         return $this->returnTaskJson($exporter->createExportTask());
     }
+
+    public function state()
+    {
+        $test =null;
```
Comment (zagovorichev) [inline controller/Results.php]: do you need this?

File: controller/ResultApi.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
+<?php
+/**
```
Comment (zagovorichev) [inline controller/ResultApi.php]: copyright

File: controller/ResultApi.php [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+use oat\taoDelivery\model\execution\DeliveryExecution;
+use oat\taoDelivery\model\execution\ServiceProxy;
+use oat\taoOutcomeUi\model\ResultsService;
+use Renderer;
+
+class ResultApi extends \tao_actions_RestController
+{
+    public function state()
+    {
+        $test = null;
```
Comment (zagovorichev) [inline controller/ResultApi.php]: why do you need this?

File: controller/Results.php [hunk 4]
```diff
@@ -31,6 +31,7 @@
 use oat\tao\model\accessControl\AclProxy;
 use oat\tao\model\plugins\PluginModule;
 use oat\tao\model\taskQueue\TaskLogActionTrait;
+use oat\taoDelivery\model\execution\DeliveryExecution;
```
Comment (zagovorichev) [inline controller/Results.php]: you've moved the logic, so you don't need to change this file

**Comment by jbout** [comment]
This functionality is tightly coupled with a specific customer score reporting and should not be part of the generic outcome ui extension.
* This functionality needs either to to be moved or generalised
* Delivery execution aliases as defined by the lti extension should be taken into account for this endpoint
* OpenApi documentation eeds to be provided as json, in order to be automatically parsed and added to the documentation, please see @dietertaotesting

File: doc/openapi.yaml [hunk 5]
```diff
@@ -0,0 +1,117 @@
+openapi: 3.0.1
```
Comment (SashaTAO) [inline doc/openapi.yaml]: Swagger documentation should be validated with @dietertaotesting to make sure they get parsed corectly

Comment (dietertaotesting) [inline doc/openapi.yaml]: At this point, the hub supports only JSON files, that live in `doc` and are named 'rest.json', 'restapi.json', 'swaggers.json', or 'swagger.json'.
This is very likely to change, but now there is that limitation.

**Pull request merged**